src/services/expensesService.js

```diff
@@ -0,0 +1,67 @@
+const { Op } = require('sequelize');
+const { models } = require('../models/models.js');
+const { Expense } = models;
+
+const getAllExpenses = async (userId, categories, from, to) => {
+  const whereClause = {};
+
+  if (userId) {
+    whereClause.userId = userId;
+  }
+
+  if (categories && categories.length > 0) {
+    whereClause.category = categories;
+  }
+
+  if (from && to) {
+    whereClause.spentAt = { [Op.between]: [from, to] };
+  } else if (from) {
+    whereClause.spentAt = { [Op.gte]: from };
+  } else if (to) {
+    whereClause.spentAt = { [Op.lte]: to };
+  }
+
+  return Expense.findAll({
+    where: whereClause,
+  });
+};
```

Comment:
This function could benefit from some error handling. Consider adding a try/catch block to handle any potential errors that might arise when interacting with the database. This will help to ensure that your application is robust and can handle unexpected situations gracefully.